### PR TITLE
scheduler: inject a failure by pod annotation

### PR DESCRIPTION
**Pull request by gregwebs**
This can be useful for testing scheduling failures.
You can for example add it to monitor-deployment
along with changing the scheduler to the custom scheduler.

### Check List 

Tests

 - Manual test (add detailed scripts or steps below)
I tested as mentioned in the PR. I observed that no other pods had failed scheduling, including after deploying a second TiDB cluster.

Code changes

 - Has Go code change

Side effects

Related changes

### Does this PR introduce a user-facing change?:
 
 ```release-note
NONE
 ```

### CODE REVIEW

File: pkg/scheduler/scheduler.go [hunk 1]
```diff
@@ -118,6 +123,15 @@ func (s *scheduler) Filter(args *schedulerapiv1.ExtenderArgs) (*schedulerapiv1.E
 	}, nil
 }
 
+// FailureError is returne when the FailTiDBSchedulerLabelKey is seen
```
Comment (tennix) [inline pkg/scheduler/scheduler.go]:
```suggestion
// FailureError is returned when the FailTiDBSchedulerLabelKey is seen
```

File: pkg/label/label.go [hunk 2]
```diff
@@ -65,6 +65,9 @@ const (
 	TiDBLabelVal string = "tidb"
 	// TiKVLabelVal is TiKV label value
 	TiKVLabelVal string = "tikv"
+	// FailTiDBSchedulerLabelKey is for injecting a failure into the TiDB custom scheduler
+	// A pod with this label will produce an error when scheduled.
+	FailTiDBSchedulerLabelKey string = "tidb.pingcap.com/fail-scheduler"
```
Comment (aylei) [inline pkg/label/label.go]: How about using annotation?

Comment (tennix) [reply pkg/label/label.go]: Yeah, I think annotation is better.

Comment (gregwebs) [reply pkg/label/label.go]: :+1: Done

**Pull request merged**